### ISSUE TITLE
OUPS was expecting to open a user friendly windows and not serialize the stack in the PharoDebug.log

### ISSUE THREAD

**Issue by VincentBlondeau** [guillep/pharo-core#7782]
**Describe the bug**
The image froze for a unknown issue and I got this log trace:
[PharoDebug.log](https://github.com/pharo-project/pharo/files/5572563/PharoDebug.log)

**To Reproduce**
N/A

**Expected behavior**
I was expecting a GUI to open to be able to recover gracefully.

**Screenshots**
N/A

**Version information:**
WIN10
Pharo 9.0.0
Build information: Pharo-9.0.0+build.855.sha.1fceb301a1b64ab952b6d7a6133e49bb0657c25d (64 Bit)

**Expected development cost**
1 day (I also think there is a Font FFI issue...)

**Additional context**
Generated in Lifeware image

**Comment by Ducasse** [comment]
Thanks vincent. Yes oups is still a bit green. I tagged Steven.

**Comment by StevenCostiou** [comment]
I'm sorry @VincentBlondeau I do not see the relation with OUPS:
1) The issue is unknown (so not reproducible)
2) The stack trace shows stuff about fonts...

@Ducasse remember we removed the emergency evaluator from the image and we replaced it with logging in PharoDebug.log. So if the system has a critical error that prevents the opening of all debuggers, OUPS now logs in that log file. 

That is what would happen if you had a bug with fonts: we won't be able to display a debugger so it would go to the log. So OUPS seems ok here.

The freeze could be an error of yours. It could also be that, if a font problem prevents the opening of all debuggers, including the Emergency Debugger, it is possible that since you need fonts everywhere each time the system tries to display something, it also generates an error and tries opening a debugger, fails because of fonts, then logs, etc. That would be the same with or without OUPS in case of such font problem.

Now, I see nothing that I can exploit to reproduce the bug, nor to make me think that it could be OUPS. Please, I need more information to help, could you:
- change the title of your issue: we do not know it's OUPS, there's nothing to think it is
- provide more details: what is the bug? what to do to reproduce it?

**Comment by StevenCostiou** [comment]
Note: a nice future improvement of OUPS would be if we could fallback to a console with REPL. There, we would not depend on anything graphical from the image/VM: no fonts, no morphic, no spec, just the debugger infrastructure and model.

**Comment by StevenCostiou** [comment]
If this one is not moving I propose to close it as is.

**Comment by VincentBlondeau** [comment]
`@Ducasse remember we removed the emergency evaluator from the image and we replaced it with logging in PharoDebug.log. So if the system has a critical error that prevents the opening of all debuggers, OUPS now logs in that log file.`

I was expecting indeed the emergency evaluator to popup. 

The issue was not about the crash by itself but the handling done by OUPS. And since the behavior changed to something I was not expecting, i.e. a log instead of a Emergency debugger, I was a bit puzzled...
